packages/grafana-data/src/transformations/matchers/nameMatcher.ts

```diff
@@ -4,32 +4,40 @@ import { FieldMatcherInfo, FrameMatcherInfo } from '../../types/transformations'
 import { stringToJsRegex } from '../../text/string';
 import { getFieldDisplayName } from '../../field/fieldState';
 
-// General Field matcher
-const fieldNameMacher: FieldMatcherInfo<string> = {
+export interface FieldNameMatcherOptions {
+  pattern: string;
+  frame?: DataFrame;
+  series?: DataFrame[];
+}
```

Comment:
Is this the correct way of naming the different "parts/levels" of the data frame? @dprokop @torkelo @ryantxu

Reply:
This I think is the wrong level, this is matcher persisted options. frame and all frames should not be part of the matcher persisted options. 

We should change the FieldMatcher function signature: https://github.com/grafana/grafana/blob/master/packages/grafana-data/src/types/transformations.ts#L28 

To include frame & allFrames

Reply:
hmm, so I totally missed the view of things being persisted. Then it makes sense to move it. The reason why I passed it in the options was that the `FieldMatcher` is being used in other scenarios e.g. `FieldTypeMatcher` and the frame and all frames doesn't make sense in those scenarios.

But I will see if I can introduce a `FieldNameMatcher` that takes the correct signature instead. It will also be more performant since it won't need to create a new option for each iteration which is good.

Reply:
hmm - so the pattern in place isn't made to be extended in that way.